src/lib.rs

```diff
@@ -28,24 +31,27 @@ decl_module! {
 		fn deposit_event() = default;
 
 		/// Updates the attestation
+		#[weight = 1]
 		pub fn update_accumulator(origin, accumulator: Vec<u8>) -> DispatchResult {
 			let attester = ensure_signed(origin)?;
 
-			let counter = if !<AccumulatorCount<T>>::exists(&attester) {
-				0
-			} else {
-				<AccumulatorCount<T>>::get(&attester)
-			};
-
-			let next = counter.checked_add(1).ok_or("Overflow increasing accumulator index")?;
-			ensure!(!<AccumulatorList<T>>::exists((&attester, next)),
-					"Inconsistent accumulator counter");
+			// if attester didn't store any accumulators, this will be 0
+			// 0 is the default value for new keys.
+			let counter = <AccumulatorCount<T>>::get(&attester);
 
-			<AccumulatorList<T>>::insert((&attester, counter), &accumulator);
-			<AccumulatorCount<T>>::insert(&attester, next);
+			// new counter value
+			let next = counter.checked_add(1).ok_or("accumulator overflow")?;
 
-			Self::deposit_event(RawEvent::Updated(attester, next, accumulator));
-			Ok(())
+			// set bytes at index `counter` to accumulator
+			// update counter to `next`
+			if !<AccumulatorList<T>>::contains_key((attester.clone(), counter)) {
+				<AccumulatorList<T>>::insert((attester.clone(), counter), &accumulator);
+				<AccumulatorCount<T>>::insert(&attester, next);
+				Self::deposit_event(RawEvent::Updated(attester, next, accumulator));
+				Ok(())
+			} else {
+				Err(dispatch::DispatchError::Other("inconsistent accumulator counter"))
```

Comment:
we could import DispatchError. Should be no problem and make the line shorter ;)